ashley/src/com/badlogic/ashley/core/Engine.java

```diff
@@ -432,8 +432,12 @@ private void processComponentOperations() {
 			ComponentOperation operation = componentOperations.get(i);
 
 			switch(operation.type) {
-				case Add: operation.entity.addInternal(operation.component); break;
```

Comment:
Components are immediately added/removed. The only thing left to do (asynchronously) is listener notification and family membership update.